backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/invoicecandidate/C_Invoice_Candidate_StepDef.java

```diff
@@ -152,8 +156,11 @@ public void recompute_invoice_candidates_if_required(@NonNull final DataTable da
 
 			if (recompute)
 			{
-				final InvoiceCandidateIdsSelection onlyInvoiceCandidateIds = InvoiceCandidateIdsSelection.ofIdsSet(
-						ImmutableSet.of(InvoiceCandidateId.ofRepoId(invoiceCandidate.getC_Invoice_Candidate_ID())));
+				final ImmutableSet<InvoiceCandidateId> invoiceCandidateIds = ImmutableSet.of(InvoiceCandidateId.ofRepoId(invoiceCandidate.getC_Invoice_Candidate_ID()));
+
+				invoiceCandDAO.invalidateCandsFor(invoiceCandidateIds);
+
```

Comment:
dev-note: normally, we don't need this for our cases due to `de/metas/invoicecandidate/async/spi/impl/InvoiceCandWorkpackageProcessor.java:121` i.e.:

```JAVA
invoiceCandDAO.invalidateCands(candidatesOfPackage);
```

but there might be cases in the future where we would use this step def to recompute candidates at a different timing so I think it makes sense to have it

Reply:
I think i don't get it...this seems to be the only *relevant* change.
if we don't need it now, then there is no change at all to fix the flakyness.
And *if* we need it, it AFAIS means that the ICs are incorrect in metasfresh, unless a cucmber-test or something unexpected happens to invalidate them.
In other words, this is actually the flakyness i'm looking forward to be settled.

In that context i now see the "whole" recompute step much more critical. It seems to indicate that the whole thing might fail and we might have wrong data in the production-ICs, unless something unrelated causes them to be recomputed.

Brainfart 1: maybe the recomputing is going on in parallel. In that case, this step needs to make sure to wait for it to end. That can happen by either waiting for and corresponding WPs to be finished, or by waiting for all C_Invoice_Candidate_Reecpute records to be gone.